src/MashStepTableModel.cpp

```diff
@@ -86,6 +86,23 @@ void MashStepTableModel::setMash( Mash* m )
    }
 }
 
+void MashStepTableModel::reorderMashSteps()
+{
+   if( mashObs && steps.size() > 0)
+   {
+      // I am NOT disconnecting the signals. I just want things redrawn
+      beginRemoveRows( QModelIndex(), 0, steps.size()-1 );
+      // Remove mashObs and all steps.
+      steps.clear();
+      endRemoveRows();
+
+      QList<MashStep*> tmpSteps = mashObs->mashSteps();
+      beginInsertRows( QModelIndex(), 0, tmpSteps.size()-1 );
+      steps = tmpSteps;
+      endInsertRows();
```

Comment:
Instead of removing and adding the steps back, could you just call

    beginResetModel();
    endResetModel();

?